File: Python Interpreter/PythonInterpreter/Block.py
class Block:

    def __init__(self):
        self.statementList = []

    def add_statement(self, stmnt):
        if stmnt is not None:
            self.statementList.append(stmnt)
        else:
            raise AttributeError("Invalid statement")

    def evaluate(self):
        for statements in self.statementList:
            statements.evaluate()

File: JavaInterpreter/src/CPLJavaInterpreter/Block.java
package CPLJavaInterpreter;

import java.util.ArrayList;
import java.util.List;

public class Block
{
    private List<Statement> statementList;

    public Block ()
    {
        statementList = new ArrayList<>();
    }

    public void addStatement(Statement stmnt)
    {
        if (stmnt != null)
            statementList.add(stmnt);
        else
            throw new IllegalArgumentException("Invalid statement");
    }

    public void evaluate()
    {
        for ( Statement s : statementList)
        {
            s.evaluate();
        }
    }
}
